core/trino-main/src/main/java/io/trino/operator/scalar/DataSizeFunctions.java

```diff
@@ -133,4 +172,140 @@ public static Unit parse(String unitString, String dataSize)
             }
         }
     }
+
+    private enum UnitDecimal
+    {
+        BYTE(BigDecimal.ONE),
+        KILOBYTE(new BigDecimal(1_000L)),
+        MEGABYTE(new BigDecimal(1_000_000L)),
+        GIGABYTE(new BigDecimal(1_000_000_000L)),
+        TERABYTE(new BigDecimal(1_000_000_000_000L)),
+        PETABYTE(new BigDecimal(1_000_000_000_000_000L)),
+        EXABYTE(new BigDecimal(1_000_000_000_000_000_000L)),
+        ZETTABYTE(new BigDecimal(1_000_000_000_000_000_000L).multiply(new BigDecimal(1_000L))),
+        YOTTABYTE(new BigDecimal(1_000_000_000_000_000_000L).multiply(new BigDecimal(1_000_000L)));
```

Comment:
`new BigDecimal("1" + "0".repeat(24))` could be more readable?
the enum constructor could take the exponent only (24), so we do this once.

Reply:
I added a `base` argument to the Unit constructor and merged decimals into it. The overall class looks better but PTAL at the constructor.